homeassistant/components/tplink/entity.py

```diff
@@ -20,10 +21,14 @@
 def async_refresh_after(
     func: Callable[Concatenate[_T, _P], Awaitable[None]],
 ) -> Callable[Concatenate[_T, _P], Coroutine[Any, Any, None]]:
-    """Define a wrapper to refresh after."""
+    """Define a wrapper to catch exceptions and refresh after calls."""
 
     async def _async_wrap(self: _T, *args: _P.args, **kwargs: _P.kwargs) -> None:
-        await func(self, *args, **kwargs)
+        try:
+            await func(self, *args, **kwargs)
+        except SmartDeviceException as ex:
+            raise HomeAssistantError("Unable to communicate with the device") from ex
```

Comment:
I think we should pass the `ex` as a parameter into the `HomeAssistantError`.  That way the log message will show us more detail of the underlying error to aid debugging and will already include some kind of "Unable to communicate" message.  So maybe the message here can say `f"Error calling {entity, func_name}"`.

Also should we be raising `IntegrationError` instead of the base class?